tests/testthat/test-strategy-k93.R

```diff
@@ -1,79 +1,103 @@
-  expect_identical(p$strategy, s)
-
-  ## Set the height to something (here 10)
-  h0 <- 10
-  p$set_state("height", h0)
-
-
-  expect_identical(p$state("height"), h0)
-
-  ## Check: Is this redundant now
-  ## We now use 
-  vars <- p$internals
-  expect_identical(p$state("height"), vars$states[which(p$ode_names == "height")])
-})
-
-test_that("Critical Names", {
-  s <- K93_Strategy()
-  my_names <- K93_Individual(s)$ode_names
-  expect_identical(my_names[1:3], c("height", "mortality", "fecundity"))
-})
-
-test_that("K93_Strategy hyper-parameterisation", {
-  s <- K93_Strategy()
-
-  # Possibly update to be relevant?
-
-  # b_0 <- c(0.059)
-  #ret <- K93_hyperpar(trait_matrix(b_0, "b_0"), s)
-
- #s expect_true(all(c("b_0", "b_1", "b_2") %in% colnames(ret)))
-  #expect_equal(ret[, "b_0"], b_0)
-  #expect_equal(ret[, "b_0"], c(1.46678,0.028600), tolerance=1e-5)
-
-  ## Empty trait matrix:
-  ret <- K93_hyperpar(trait_matrix(numeric(0), "b_0"), s)
-  expect_equal(ret, trait_matrix(numeric(0), "b_0"))
-})
-
+# Built from  tests/testthat/test-strategy-ff16r.R on Wed Aug 12 15:33:08 2020 using the scaffolder, from the strategy:  FF16r
+# Built from  tests/testthat/test-strategy-ff16.R on Fri Jul  3 08:14:35 2020 using the scaffolder, from the strategy:  FF16
+context("Strategy-K93")
+
+test_that("Defaults", {
+  expected <- list(
+   height_0 = 2.0,                   
+   b_0 = 0.059,
+   b_1 = 0.012,
+   b_2 = 0.00041,
+   c_0 = 0.008,
+   c_1 = 0.00044,
+   d_0 = 0.00073,
+   d_1 = 0.044,
+   S_D = 1,
+   control = Control())
+
+  keys <- sort(names(expected))
+
+  s <- K93_Strategy()
+  expect_is(s, "K93_Strategy")
+
+  expect_identical(sort(names(s)), keys)
+  expect_identical(unclass(s)[keys], expected[keys])
+})
+
+test_that("K93 collect_all_auxillary option", {
+
+  s <- K93_Strategy()
+  p <- K93_Individual(s)
+  expect_equal(p$aux_size, 1)
+  expect_equal(length(p$internals$auxs), 1)
+  expect_equal(p$aux_names, c(
+    "competition_effect"
+  ))
+})
+
+test_that("Reference comparison", {
+  s <- K93_Strategy()
+  p <- K93_Individual(s)
+
+  expect_identical(p$strategy, s)
+
+  ## Set the height to something (here 10)
+  h0 <- 10
+  p$set_state("height", h0)
+
+  expect_identical(p$state("height"), h0)
+
+  ## Check: Is this redundant now
+  ## We now use 
+  vars <- p$internals
+  expect_identical(p$state("height"), vars$states[which(p$ode_names == "height")])
+})
+
+test_that("Critical Names", {
+  s <- K93_Strategy()
+  my_names <- K93_Individual(s)$ode_names
+  expect_identical(my_names[1:3], c("height", "mortality", "fecundity"))
+})
+
+test_that("K93_Strategy hyper-parameterisation", {
+  s <- K93_Strategy()
+  
+  ## Hyperpars should just pass through:
+  ret <- K93_hyperpar(trait_matrix(numeric(0), "b_0"), s)
+  expect_equal(ret, trait_matrix(numeric(0), "b_0"))
+})
+
+
+test_that("K93 seed rain is unchanged", {
+  
+  # Generic parameters
+  p0 <- scm_base_parameters("K93")
+  p0$k_I <- 1e-6
+  p0$disturbance_mean_interval <- 200

```

Comment:
The long disturbance regime (200 years) causes this to run very slowly - 240s. Ideally tests would take no longer than 10s. Until we identify the cause of the slowdown, could we  change to an example with interval of 10 years, so that tests run quickly?

Perhaps leave the old one in there but commented out?

The example won't match the one from paper, but that's ok. As long as we can detect if the output of the model changes from what it is currently, that will be sufficient.

Reply:
Good point! I was lazy and just re-used the model parameters, but it's muuuch faster now.